mangaki/mangaki/tests/test_anidb.py

```diff
@@ -47,5 +52,19 @@ def test_anidb_get(self):
             content_type='application/xml'
         )
         anime = self.anidb.get_or_update_work(11606)
-        self.assertNotEqual(anime.title, '')
-        self.assertNotEqual(anime.worktitle_set.count(), 0)
+
+        self.assertEqual(anime.title, 'Sangatsu no Lion')
+        self.assertEqual(anime.nb_episodes, 22)
+        self.assertEqual(anime.studio.title, 'Shaft')
+
+        author_ids = [q.artist_id for q in Staff.objects.filter(work_id=anime.pk, role_id=3)]
+        author_names = [Artist.objects.get(pk=at_id).name for at_id in author_ids]
+        self.assertEqual(author_names, ['Umino Chika'])
```

Comment:
It's an absolutely awesome write-up!

I think you're most likely right for your conclusion, the last way I see to fetch the names is to filter before `all()` by `role__slug__in=('author', 'composer', 'director')` and add `role__slug` in your `values_list` so that you can differentiate the `roles` and build the different lists using `itertools.groupby` or something similar.

How did you perform your measurements of timing?
For my part, I got 6.87 ms per loop (best of 3) over 5000 iterations using [`timeit`](https://docs.python.org/3/library/timeit.html) which is often the right way to perform such measurements.